### PR TITLE
Add check probs sum to 1

### LINKED ISSUE #560: Doubt about arrows and dashed lines in directed PAGA
Hi

I've got a question about the directed PAGA. I see that different clusters are joined by arrows or dashed lines, but I didn't find any info on their meaning in cellrank docs. Could you explain what's their difference? What does two clusters connected with a thick dashed line imply?

image

image

Thanks for your help!

### PR DESCRIPTION
**Pull request by michalk8**
**IMPORTANT: Please search among the [Pull request](../pulls) before creating one.**

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] New feature (non-breaking change which adds functionality)

## Description

Raise `ValueError` of absorption probabilities do not sum to 1 in `BaseEstimator`.

## How has this been tested?

2 new tests added.

## Closes

closes #560

### CODE REVIEW

File: cellrank/tl/estimators/_base_estimator.py [hunk 1]
```diff
@@ -517,6 +517,21 @@ def compute_absorption_probabilities(
...
+                f"specifying a preconditioner as `preconditioner=...` or "
+                f"use a direct solver as `solver='direct'` if the matrix is small."
+            )
+        mask = np.isclose(abs_classes.sum(1), 1.0, rtol=1e-3)
+        if not np.all(mask):
+            raise ValueError(
+                f"`{np.sum(~mask)}` value(s) do not sum to 1 (rtol=1e-3). Try decreasing the tolerance as `tol=...`, "
+                f"specifying a preconditioner as `preconditioner=...` or "
+                f"use a direct solver as `solver='direct'` if the matrix is small."
+            )
```
Comment (Marius1311) [inline cellrank/tl/estimators/_base_estimator.py:533]: This looks good to me.

File: tests/test_cflare.py [hunk 2]
```diff
@@ -590,6 +590,55 @@ def test_manual_approx_rc_set(self, adata_large):
...
+            "cellrank.tl.estimators._base_estimator._solve_lin_system",
+            return_value=abs_prob,
+        )
+
+        with pytest.raises(
+            ValueError, match=r"`1` value\(s\) do not sum to 1 \(rtol=1e-3\)."
+        ):
+            mc.compute_absorption_probabilities()
+
+    def test_abs_probs_do_not_sum_to_1(self, adata_large: AnnData, mocker):
```
Comment (Marius1311) [inline tests/test_cflare.py:618]: Naming confusing. Could it be that this test checks for negative values, and the one above checks for summation to one?

File: tests/test_cflare.py [hunk 3]
```diff
@@ -590,6 +590,55 @@ def test_manual_approx_rc_set(self, adata_large):
...
+        mc.set_terminal_states(
+            {"x": adata_large.obs_names[:3], "y": adata_large.obs_names[3:6]}
+        )
+
+        n_term = np.sum(~pd.isnull(mc.terminal_states))
+        abs_prob = np.zeros((adata_large.n_obs - n_term, n_term))
+        abs_prob[:, 0] = 1.0
+        abs_prob[0, 0] = -0.5
+        abs_prob[0, 1] = -1.5
+        mocker.patch(
```
Comment (Marius1311) [inline tests/test_cflare.py:634]: Yes, looks like this one tests for negative values, no?

**Review by Marius1311 — COMMENTED** [review]
Looks good! Some test namings are not clear to me.

File: tests/test_cflare.py (hunk 3 above)
Comment (michalk8) [reply tests/test_cflare.py:634]: Yes, thanks for spotting this, fixed.

File: tests/test_cflare.py (hunk 2 above)
Comment (michalk8) [reply tests/test_cflare.py]: Fixed, thanks.

**Comment by michalk8** [comment]
The discussion whether to add a preconditioner will continue in #564 , merging this.

**Pull request merged**